src/app/_components/tasks/LocationModal.tsx

```diff
@@ -0,0 +1,141 @@
+'use client';
+
+import Image from 'next/image';
+import { Dispatch, SetStateAction, useRef, useState } from 'react';
+import NormalInput from '../common/NormalInput';
+import SubmitButton from '../common/SubmitButton';
+import { searchLocation } from '@/app/_apis';
+import useOutsideClick from '@/app/_hooks';
+
+interface LocationModalProps {
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  setPlaceName: (name: string) => void;
+}
+
+interface SearchResult {
+  title: string;
+  address: string;
+  category: string;
+  description: string;
+  link: string;
+  mapx: string;
+  mapy: string;
+  roadAddress: string;
+  telephone: string;
+}
+
+function LocationModal({ setIsOpen, setPlaceName }: LocationModalProps) {
+  const locationRef = useOutsideClick(() => setIsOpen(false));
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [searchPlace, setSearchPlace] = useState('');
+  const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+
+  const handleCloseButton = () => {
+    setIsOpen(false);
+  };
+
+  const handleSearchButton = async () => {
+    if (!searchPlace.trim()) {
+      setSearchPlace('');
+      setSearchResults([]);
+      return;
+    }
+
+    try {
+      const results = await searchLocation(searchPlace);
+
+      if (results) {
+        setSearchResults(results.data);
+      } else {
+        console.error('API 호출에 실패했습니다.');
+      }
```

Comment:
얼리 리턴으로 처리해도 좋을 것 같아요!

```suggestion
      if (!results) return console.error('API 호출에 실패했습니다.');
      
      setSearchResults(results.data);
```

Reply:
이렇게 하니까 더 깔끔하네요! 얼리 리턴 적용해 두겠습니다 ㅎㅎ